experimental/lib/Transforms/Speculation/HandshakeSpeculation.cpp

```diff
@@ -219,10 +219,20 @@ LogicalResult HandshakeSpeculationPass::routeCommitControl() {
 
   llvm::DenseSet<Operation *> arrived;
   std::vector<BranchTracingItem> branchTrace;
+  // Start traversal from the speculator
   for (OpOperand &succOpOperand : specOp.getDataOut().getUses()) {
     routeCommitControlRecursive(&getContext(), specOp, arrived, succOpOperand,
                                 branchTrace);
   }
+  // Start traversal from save-commit units
+  for (auto saveCommitOp :
+       mlir::cast<FuncOp>(specOp->getParentOp()).getOps<SpecSaveCommitOp>()) {
+    for (OpOperand &succOpOperand : saveCommitOp.getDataOut().getUses()) {
+      branchTrace.clear();
+      routeCommitControlRecursive(&getContext(), specOp, arrived, succOpOperand,
+                                  branchTrace);
+    }
+  }
```

Comment:
@murphe67 This can't be replaced with placements, I missed that the save-commit and commit units might be placed at the same location. In that case, the operand used for placement is located below both units.

Instead of using the connection from the speculator to collect all save-commits, I used `getOps`. This avoids introducing a dependency on the save-commit controls being routed first.